src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,102 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private Bucket<K, V>[] table;
+    private int size = 0;
+
+    public MyHashMap() {
+        table = new Bucket[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
+        checkCapacity();
+        int position = getPosition(key);
+        Bucket<K, V> bucket = table[position];
+        Bucket<K, V> newBucket = new Bucket<>(key, value);
+        if (table[position] == null) {
+            table[position] = newBucket;
+        } else {
+            if (Objects.equals(bucket.key, newBucket.key)) {
+                bucket.value = newBucket.value;
+                return;
+            }
+            Bucket<K, V> tempBucket = bucket;
+            while (tempBucket.next != null) {
+                tempBucket = tempBucket.next;
+                if (Objects.equals(tempBucket.key, newBucket.key)) {
+                    tempBucket.value = newBucket.value;
+                    return;
+                }
+            }
+            tempBucket.next = newBucket;
+        }
+        size++;
+    }
 
+    private int getPosition(K key) {
+        int position;
+        if (key == null) {
+            position = 0;
+        } else {
+            position = Math.abs(key.hashCode() % table.length);
+        }
+        return position;
+    }
+
+    private void checkCapacity() {
+        int capacity = table.length;
+        Bucket<K, V> temp;
+        if (size > (int) (table.length * LOAD_FACTOR)) {
+            capacity = capacity << 1;
+            Bucket<K, V>[] oldTable = table;
+            table = (Bucket<K, V>[]) new Bucket[capacity];
+            size = 0;
+            for (Bucket<K, V> kvBucket : oldTable) {
+                temp = kvBucket;
+                while (temp != null) {
+                    put(temp.key, temp.value);
+                    temp = temp.next;
+                }
+            }
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int position;
+        position = getPosition(key);
+        Bucket<K, V> bucket = table[position];
+        if (bucket == null) {
+            return null;
+        } else {
```

Comment:
useless block, you'll return `null` in the end of method, since your condition in loop will be `false`